### ISSUE TITLE
Picture download memory explosion

### ISSUE THREAD

**Issue by baneyue** [Baseflow/flutter_cached_network_image#663]
## 🐛 Bug Report

When I am on Windows platform, listview has 7000 pieces of data and loads pictures. The size of each picture is about 100kb. I see that disk cache is not enabled and the network speed instantly reaches 30-60mb / s.

Whether the concurrent download queue and the number of concurrent downloads can be manually set;

Whether disk cache can be provided;

At present, I feel that there are still basic performance problems with this library.

### Configuration

**Version:** 3.1.0

**Platform:** 
- [ ] :iphone: Windows

**Comment by baneyue** [comment]
Is there any progress on this issue

It's really a big problem

**Comment by baneyue** [comment]
When you quickly drag the scroll bar, the network speed will soon reach more than 20m / s, and then it will be killed as an exception by the anti-virus software. I mean, the number of pictures loaded in parallel is too large, and the control of rolling out of the screen has not been cancelled.

![企业微信截图_16364515438502](https://user-images.githubusercontent.com/1902581/140902082-278e21a2-5c49-4d68-b426-ffc13299fd54.png)

**Comment by anirudhsharma392** [comment]
for now lowering the value of memCacheHeight helped me, but it's a temporary fix !

**Comment by renefloor** [comment]
@baneyue is this really different when you use NetworkImage instead of CachedNetworkImage?

**Comment by baneyue** [comment]
yes , The performance problem is the biggest problem of cachednetworkimage, and there is no good solution at present

**Comment by Klabauterman** [comment]
I am having the same problem. I am even getting out of memory exceptions.

**Comment by boomery** [comment]
same problem, do you got any solution for this ?
